.github/workflows/triggered_target_build.yml

```diff
@@ -0,0 +1,117 @@
+name: Trigger target build of docs
+# this workflow is triggered by a comment on a PR or manually via workflow_dispatch
+# It allows you to specify a make target for building the docs for a branch/PR
+# When triggered by a comment, it will run on the PR branch and report status on the PR
+# # This is a one-time-run, it does not run on every push to the PR branch
+
+on:
+  issue_comment:
+    types: [created]
+  workflow_dispatch:
+    inputs:
+      make_target:
+        description: "Enter make target: html html-noplot docs slimfast slimgallery"
+        type: string
+        default: "slimfast"
+
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
+  statuses: write
+
+jobs:
+  determine-make-target:
+    runs-on: ubuntu-latest
+    outputs:
+      target: ${{ steps.determine-make-target.outputs.target }}
+      pr_ref: ${{ steps.get-pr-info.outputs.pr_ref }}
+      pr_sha: ${{ steps.get-pr-info.outputs.pr_sha }}
```

Comment:
These are the 3 key things we need to know about the trigger:
- what make target
- pr details, so we can get the right stuff to build and then report back to the right status.